### PR TITLE
Struct property sorting

### PR DESCRIPTION
**Pull request by jasdom**
Changed struct properties to be sorted.

### CODE REVIEW

File: packages/components/src/presentation-components/common/ContentBuilder.ts [hunk 1]
```diff
@@ -139,9 +139,10 @@ class StructMembersAppender implements INestedPropertiesAppender {
     this._members[record.fieldHierarchy.field.name] = record.record;
   }
   public finish(): void {
+    const sortedMembers = Object.entries(this._members).sort();
     const value: StructValue = {
       valueFormat: UiPropertyValueFormat.Struct,
-      members: this._members,
+      members: Object.fromEntries(sortedMembers),
```
Comment (saskliutas) [inline packages/components/src/presentation-components/common/ContentBuilder.ts:145]: Can you add test to check properties order?

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: packages/components/src/presentation-components/common/ContentBuilder.ts (hunk 1 above)
Comment (jasdom) [reply packages/components/src/presentation-components/common/ContentBuilder.ts:145]: Added.

File: packages/components/src/presentation-components/common/ContentBuilder.ts [hunk 2]
```diff
@@ -139,9 +139,10 @@ class StructMembersAppender implements INestedPropertiesAppender {
     this._members[record.fieldHierarchy.field.name] = record.record;
   }
   public finish(): void {
+    const sortedMembers = Object.entries(this._members).sort();
```
Comment (grigasp) [inline packages/components/src/presentation-components/common/ContentBuilder.ts:142]: In what order is this going to sort? `Object.entries` returns an array of tuples and `sort` orders items in ascending ASCII order 🤔 I think in the best case scenario it's going to be by the first tuple item - the property name, which is not necessarily correct. We want to sort by property label instead.

**Review by grigasp — CHANGES_REQUESTED** [review]

**Comment by grigasp** [comment]
Please also add a changeset.

File: packages/components/src/presentation-components/common/ContentBuilder.ts (hunk 2 above)
Comment (jasdom) [reply packages/components/src/presentation-components/common/ContentBuilder.ts]: It was sorting based on the property name, changed to sort by label instead.

File: .changeset/gorgeous-plants-nail.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/presentation-components": patch
+---
+
+Added struct member sorting.
```
Comment (grigasp) [inline .changeset/gorgeous-plants-nail.md:5]: Should try to be more specific.

```suggestion
Sort struct property members by label when creating content for property grid, table, and other content components.
```

Comment (jasdom) [reply .changeset/gorgeous-plants-nail.md]: Updated the description.

**Review by saskliutas — APPROVED** [review]

**Review by grigasp — APPROVED** [review]

**Pull request merged**